spec/workers/csv_export_worker_spec.rb

```diff
@@ -0,0 +1,111 @@
+require 'govuk_sidekiq/testing'
+
+RSpec.describe CsvExportWorker do
+  let(:search_params) do
+    {
+      date_range: 'past-30-days',
+      organisation_id: 'org-id',
+      document_type: 'news_story'
+    }
+  end
+
+  let(:document_types) do
+    [{ id: 'case_study', name: 'Case study' },
+     { id: 'guide', name: 'Guide' },
+     { id: 'news_story', name: 'News story' },
+     { id: 'html_publication', name: 'HTML publication' }]
+  end
+
+  let(:organisations) do
+    [{ name: 'org', id: 'org-id', acronym: 'OI', },
+     { name: 'another org', id: 'another-org-id', },
+     { name: 'Users Org', id: 'users-org-id', acronym: 'UOI', }]
+  end
+
+  let(:content_items) do
+    [
+      {
+        title: 'GOV.UK homepage',
+        base_path: '/',
+        organisation_id: nil,
+        document_type: 'homepage',
+        upviews: 15,
+        pviews: 25,
+        satisfaction: 0.25,
+        useful_yes: 100,
+        useful_no: 300,
+        searches: 14,
+        feedex: 24,
+        word_count: 50,
+        reading_time: 50,
+        pdf_count: 0
+      },
+      {
+        title: 'Title 1',
+        base_path: '/base-path-1',
+        organisation_id: 'another-org-id',
+        document_type: 'guide',
+        upviews: 15,
+        pviews: 25,
+        satisfaction: 0.6,
+        useful_yes: 300,
+        useful_no: 200,
+        searches: 14,
+        feedex: 24,
+        word_count: 100,
+        reading_time: 100,
+        pdf_count: 3
+      }
+    ]
+  end
+
+  before do
+    Sidekiq::Worker.clear_all
+
+    allow(FetchDocumentTypes).to receive(:call)
+      .and_return(document_types: document_types)
+    allow(FetchOrganisations).to receive(:call)
+      .and_return(organisations: organisations)
+    allow(FindContent).to receive(:enum)
+      .with(search_params)
+      .and_return(content_items)
+
+    Fog.mock!
```

Comment:
As per my comment above we would not need mock all this requests as they can be tested in a unit test.

Reply:
Still need to mock this for the worker as will ultimately call S3.

Reply:
@theseanything: Sorry if I wasn't clear on my previous message: in the unit test you could mock the method of the class that you have extracted (you own the API), considering that:

1. The unit test for the class that you have extracted would do the testing
2. The feature test would not mock anything